subsys/bluetooth/host/Kconfig

```diff
@@ -351,6 +351,24 @@ config BT_OOB_DATA_FIXED
 	  value. This option should only be enabled for debugging and should
 	  never be used in production.
 
+config BT_KEYS_OVERWRITE_OLDEST
+	bool "Overwrite oldest keys with new ones if key storage is full"
+	help
+	  With this option enabled, if a pairing attempt occurs and the key storage
+	  is full, then the oldest keys in storage will be removed to free space
+	  for the new pairing keys.
+
+if BT_KEYS_OVERWRITE_OLDEST
+config BT_KEYS_OVERWRITE_OLDEST_CNTR_IN_SETTINGS
+	bool "Owerwrite aging counter will be stored in settings explicitly"
+	depends on BT_SETTINGS
```

Comment:
Would it make sense to make this `default y` so we guarantee correct behaviour? Only if the user knows that it can trust implicit ordering by the settings backend should they go and disable this option.